File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int zeroIndex = 0, oneIndex = 0, twoIndex = nums.size()-1;
        
        while(oneIndex <= twoIndex) {
            
            if(nums[oneIndex] == 1)
                oneIndex++;
            
            else if(nums[oneIndex] == 0)
                swap(nums[zeroIndex++], nums[oneIndex]);
            
            else if(nums[oneIndex] == 2)
                swap(nums[oneIndex], nums[twoIndex--]);
            
            if(zeroIndex > oneIndex)
                oneIndex = zeroIndex;
            
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.cs
public class Solution {
    public void SortColors(int[] nums) {
        int zeros = 0, two = nums.Length - 1, i=0;
        while (i <= two) {
            if(nums[i] == 0) {
                (nums[i], nums[zeros]) = (nums[zeros], nums[i]);
                zeros++;
                i++;
            }
            else if(nums[i] == 2) {
                (nums[i], nums[two]) = (nums[two], nums[i]);
                two--;
            }
            else {
                i++;
            }
        }
    }
}